wpilib/wpilib/powerdistributionpanel.py

```diff
@@ -98,29 +98,9 @@ def clearStickyFaults(self):
         """
         hal.clearPDPStickyFaults(self.module)
 
-    # Live Window code, only does anything if live window is activated.
-
-    def getSmartDashboardType(self):
-        return "PowerDistributionPanel"
-
-    def initTable(self, subtable):
-        if subtable is not None:
-            self.chanEntry = []
-            for channel in range(16):
-                self.chanEntry.append(subtable.getEntry("Chan" + str(channel)))
-            self.voltageEntry = subtable.getEntry("Voltage")
-            self.totalCurrentEntry = subtable.getEntry("TotalCurrent")
-            self.updateTable()
-        else:
-            self.chanEntry = None
-            self.voltageEntry = None
-            self.totalCurrentEntry = None
-
-    def updateTable(self):
-        if self.chanEntry is not None:
-            for channel in range(16):
-                self.chanEntry[channel].setDouble(self.getCurrent(channel))
-        if self.voltageEntry is not None:
-            self.voltageEntry.setDouble(self.getVoltage())
-        if self.totalCurrentEntry is not None:
-            self.totalCurrentEntry.setDouble(self.getTotalCurrent())
+    def initSendable(self, builder):
+        builder.setSmartDashboardType("PowerDistributionPanel")
+        for chan in range(self.kPDPChannels):
+            builder.addDoubleProperty("Chan%s" % (chan,), partial(self.getCurrent, chan), None)
```

Comment:
BTW, you could write `'Chan%d' % chan` instead; you know `range()` is only going to give you integers.

Reply:
force of habit

Reply:
also why add it as a tuple with one element?

Reply:
@Twinters007 what do you mean?

Reply:
`(chan,)`

Reply:
What do you mean by "add" though? I know that's a 1-tuple…

Reply:
Oh I mean add it to the string via formatting

Reply:
Common safeguard in case the single thing you're formatting in happens to be a tuple.

Reply:
Really? Huh. But we know it's gonna be an int here, no?

Reply:
Yes, thank you for repeating what I said 10 comments ago.